src/loader.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Animated loader for initial loading screen.
+ */
+export class ULabelLoader {
+    
+    public static add_loader_div(
+        container: HTMLElement
+    ) {
+        const loader_overlay = document.createElement('div');
+        loader_overlay.classList.add('ulabel-loader-overlay');
+
+        const loader = document.createElement('div');
+        loader.classList.add('ulabel-loader');
+        
+        loader_overlay.appendChild(loader);
+        container.appendChild(loader_overlay);
+        
+        ULabelLoader.add_loader_css();
+    }
+    
+    public static remove_loader_div() {
+        const loader = document.querySelector('.ulabel-loader-overlay');
+        if (loader) {
+            loader.remove();
+        }
+    }
+    
+    public static add_loader_css() {
+        const css = `
+            .ulabel-loader-overlay {
+                position: fixed;
+                width: 100%;
+                height: 100%;
+                inset: 0;
+                background-color: rgba(0, 0, 0, 0.5);
+                z-index: 100;
+            }
+            .ulabel-loader {
+                border: 16px solid #f3f3f3;
+                border-top: 16px solid #3498db;
+                border-radius: 50%;
+                width: 120px;
+                height: 120px;
+                animation: spin 2s linear infinite;
+                position: fixed;
+                inset: 0;
+                margin: auto;
+            }
+            
+            @keyframes spin {
+                0% { transform: rotate(0deg); }
+                100% { transform: rotate(360deg); }
+            }
+        `;
+        const style = document.createElement('style');
+        style.innerHTML = css;
+        document.head.appendChild(style);
```

Comment:
on the one hand i like having css close to the thing its actually styling, on the other is there a reason to attach these styles to the document and have them persist after the loader div is removed? versus we could attach the styles directly to the elements rather than making global class definitions

Reply:
Yeah we could keep track of the `<style>` element and also remove that, or maybe make it a child of the overlay?
The latter is the easiest (I think), just change `add_loader_css()` to `build_style()` that returns the `<style>` element, and let `add_loader_div()` handle where to attach it.

Reply:
I went ahead and did this in 5480b19, and confirmed that the style got removed. I like this structure better anyways.

Also in the previous version I was adding the style twice, which was probably harmless but not ideal.